docs/design-notes/raft-in-scylla.md

```diff
@@ -0,0 +1,207 @@
+Date: 2022-05-06
+
+# Intro
+
+When we began implementing Raft we wanted to create a reusable and
+well tested component which we could utilize for data, schema and topology
+operations. This is why the Raft library in raft/ has the only dependency
+- on seastar. It provides its own design documents and a README.
+
+# Raft application in Scylla
+
+In order to use the library, the client (Scylla server) needs to
+provide implementations for three key interfaces:
+    - persistence - to persist Raft state
+    - rpc - to exchange messages with instances of the library
+      on other machines
+    - the client state machine - to execute commands once
+    they were replicated and committed on a majority of nodes.
+
+Depending on the application (data, topology, or schema) Scylla can use
+separate instantiations of the library with different parameters.
+The term we use commonly for these instantiations is Raft groups.
```

Comment:
From reading this paragraph I feel one may get the (wrong) impression that there will be one separate Raft group/instantiation for each use case.

I would add a sentence which says that some applications may require multiple instantiations/groups which use the same implementations of persistence, rpc, and client state machine, and an example is data replication where we shard the set of partitions and create a separate group for each shard.